cdp/historical_balance.py

```diff
@@ -0,0 +1,92 @@
+from decimal import Decimal
+
+from cdp.asset import Asset
+from cdp.client.models.historical_balance import HistoricalBalance as HistoricalBalanceModel
+
+
+class HistoricalBalance:
+    """A class representing a balance."""
+
+    def __init__(self, amount: Decimal, asset: Asset, block_height: str, block_hash: str):
+        """Initialize the Balance class.
+
+        Args:
+            amount (Decimal): The amount.
+            asset (Asset): The asset.
+            block_height (str): the block height where the balance is in.
+            block_hash (str): the block hash where the balance is in.
+
+        """
+        self._amount = amount
+        self._asset = asset
+        self._block_height = block_height
+        self._block_hash = block_hash
+
+    @staticmethod
```

Comment:
Let's make this a `@classmethod` decorated function since we need to instantiate a class object instance.